src/WebJobs.Script/Description/Workers/WorkerFunctionDescriptorProvider.cs

```diff
@@ -90,5 +102,37 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
 
             return parameters;
         }
+
+        private bool BindingAttributeContainsExpression(IEnumerable<FunctionBinding> bindings)
+        {
+            if (bindings.Count() <= 0)
+            {
+                return false;
+            }
+
+            var bindingExpressionRegex = new Regex(@"{(.*?)\}");
+
+            foreach (ExtensionBinding binding in bindings)
+            {
+                if (binding.Metadata.IsTrigger)
+                {
+                    // skip triggers, we only care if input and output bindings contain expressions
+                    continue;
+                }
+
+                foreach (var attribute in binding.Attributes)
+                {
+                    foreach (var property in attribute.GetType().GetProperties())
```

Comment:
Should use the same binding flags that AttributeCloner uses for binding attributes https://github.com/Azure/azure-webjobs-sdk/blob/3f4ec78be9f43bb041937425ced00b341883aa42/src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs#L42. We should also filter these properties to just string properties, rather than doing ToString below. We wouldn't want to run this on any non-string properties.

Reply:
I keep ended up with a object array, so this is what I managed to get working but still seems heavy? Open to further suggestions


```csharp
var properties = attribute.GetType()
                    .GetProperties(flags)
                    .Where(prop => prop.PropertyType == typeof(string))
                    .Select(p => p.GetValue(attribute))
                    .ToArray();

foreach (var property in properties)
{
    if (_expressionRegex.IsMatch((string)property))
```

Reply:
Narrowed it down to this:

```
return attribute.GetType()
    .GetProperties(flags)
    .Any(prop => prop.PropertyType == typeof(string) && _expressionRegex.IsMatch((string)prop.GetValue(attribute)));
```